lib/firmata.js

```diff
@@ -1690,6 +1719,337 @@ Board.prototype.pingRead = function(opts, callback) {
+    this.once("stepper-done-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to report the position of a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperReportPosition = function(deviceNum, callback) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x06, // STEPPER_REPORT_POSITION from firmware
+    deviceNum,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+
+  if (callback) {
+    this.once("stepper-position-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to set the acceleration for a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ * @param {number} acceleration Desired acceleration in steps per sec^2
+ */
+
+Board.prototype.accelStepperAcceleration = function(deviceNum, acceleration) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x08, // STEPPER_SET_ACCELERATION from firmware
+    deviceNum];
+  
+  Array.prototype.push.apply(data, Board.encodeCustomFloat(acceleration));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+};
+
+
+/**
+ * Asks the arduino to set the max speed for a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ * @param {number} speed Desired speed or maxSpeed in steps per second
+ * @param {function} [callback]
+ */
+
+Board.prototype.accelStepperSpeed = function(deviceNum, speed) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x09, // STEPPER_SET_SPEED from firmware
+    deviceNum];
+  
+  Array.prototype.push.apply(data, Board.encodeCustomFloat(speed));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to configure a multiStepper group
+ * @param {object} opts Options:
+ *    {number} groupNum: Group number for the multiSteppers (range 0-5)
+ *    {number} devices: array of accelStepper device numbers in group
+ **/
+
+Board.prototype.multiStepperConfig = function(opts) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x20, // MULTISTEPPER_CONFIG from firmware
+    opts.groupNum
+  ];
+
+  Array.prototype.push.apply(data, opts.devices);
+ 
+  data.push(END_SYSEX);
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to move a multiStepper group
+ * @param {object} opts Options:
+ *    {number} groupNum: Group number for the multiSteppers (range 0-5)
+ *    {number} positions: array of absolute stepper positions
+ **/
+
+Board.prototype.multiStepperTo = function(groupNum, positions, callback) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x21, // MULTISTEPPER_TO from firmware
+    groupNum
+  ];
+
+  positions.forEach( function(position) {
+    Array.prototype.push.apply(data, Board.encode32BitSignedInteger(position));
+  });
+ 
+  data.push(END_SYSEX);
+  this.transport.write(new Buffer(data));
+
+  if (callback) {
+    this.once("multi-stepper-done-" + groupNum, callback);
```

Comment:
@dtex If the user calls `multiStepperStop`, this callback will never be handled. I'm not sure of the details of Node's EventEmitter `once` method and if it cleans up unhandled events (if the next call to `once` would deallocate memory from the previous unhandled event) or if we risk a memory leak here. One potential work around is to call reportGroupComplete in the firmware when a multiStepperStop message is received (see comment I left in ConfigurableFirmata).

Curious if @rwaldron has any insight into this as well since it's probably come up when implementing J5 modules.

Reply:
Added call to reportGroupComplete in the firmware.